vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/SpaceLinkManagerTest.java

```diff
@@ -226,4 +227,34 @@ public void test_updateLink_success() throws SpaceDoesNotExistException, LinkDoe
         Assert.assertEquals(spaceLinkDisplayUpdated.getLink().getTargetSpace(), actualUpdatedLink.getLink().getTargetSpace());
         Assert.assertEquals(spaceLinkDisplayUpdated.getType(), actualUpdatedLink.getType());
     }
+    
+    @Test
+    public void test_deleteSpaceLinkWithSourceAsNull_present() {
+        SpaceLink spaceLink = new SpaceLink();
+        spaceLink.setSourceSpace(null);
+        spaceLink.setId(spaceLinkId1);
+        List<ISpaceLink> spaceLinks =  new ArrayList<ISpaceLink>();
+        spaceLinks.add(spaceLink);
+        Mockito.when(spaceLinkRepo.findBySourceSpaceIsNull()).thenReturn(spaceLinks);
+        Mockito.doNothing().when(spaceLinkDisplayRepo).deleteByLink(spaceLink); 
+        Mockito.doNothing().when(spaceLinkRepo).deleteBySourceSpaceIdIsNull(); 
```

Comment:
why shouldn't anything be done here? also, you need to test if what the method did succeeded; "verify" is probably your best option here.

Reply:
Thanks, i have changed it to verify, my bad